### ISSUE TITLE
Get a grip on rap4-student-prototype image

### ISSUE THREAD

**Issue by hanjoosten** [AmpersandTarski/RAP#386]
Currently there is no clear version policy on the rap4-student-prototype image. This manifests itself in subtitle problems when updating and releasing RAP. Also, on inspection of the current main branch, several .yaml files refer to different versions in an inimitable way. It is no wonder that these problems exist.

**Comment by hanjoosten** [comment]
I see the problem lies even deeper. We use both Kubernetes and docker-compose. There are shared values all over the place. 
A solution to get grip on this is to work with a single file for all these shared variables (not only the ones we ask from the user). With docker-compose, this would be the `.env` file. With Kubernetes, this would be the use of [ConfigMap](https://kubernetes.io/docs/concepts/configuration/configmap/) for non-sensitive variables, and [Secret](https://kubernetes.io/docs/concepts/configuration/secret/) for confidential data. Docker Compose (v3.1 and later) supports secrets as a native feature, which is a secure way to handle sensitive data. It is recommended to not handle such data in an .env file as we still currently do. For secrets, we need to rewrite all our docker-compose.yml files to include a top level `secrets:` object as well.

**Comment by Michiel-s** [comment]
You are absolutely right. Not sure if using better docker compose constructs will help here.

We have the issue that with the new frontend the RAP4 student prototype setup won't work anymore because of the required Angular build step. So a complete new setup is needed anyway. I think RAP should focus on being the repository of ampersand projects, i.e. student scripts, with minimal functionality; e.g. generating functinonal specifications.

Generating a prototype application and deploying with ingress and database should be offloaded to student personal environments, e.g. by means of a Github codespace an reusing the project template repository for that. For students to be able to work with that the way of working needs to be specified. That's all.

We can deprecate the RAP4 student prototype image and don't have to put effort in finding a solution to upgrade to new compiler version and new frontend.
